src/Meziantou.Analyzer/Rules/AvoidUsingRedundantElseAnalyzer.cs

```diff
@@ -32,27 +33,64 @@ public override void Initialize(AnalysisContext context)
+            if (elseClause is null)
                 return;
 
-            if (!(elseClauseSyntax.Parent is IfStatementSyntax ifStatementSyntax))
+            if (!(elseClause.Parent is IfStatementSyntax ifStatement))
                 return;
 
-            var statementOrBlock = ifStatementSyntax.Statement;
-            if (statementOrBlock is null)
+            var thenStatement = ifStatement.Statement;
+            var elseStatement = elseClause.Statement;
+            if (thenStatement is null || elseStatement is null)
                 return;
 
-            var result = context.SemanticModel.AnalyzeControlFlow(statementOrBlock);
-            if (!result.Succeeded)
+            // If the 'else' clause contains a "using statement local declaration" as direct child, return
+            // NOTE:
+            //  using var charEnumerator = "".GetEnumerator();          => LocalDeclarationStatementSyntax  (will return)
+            //  using (var charEnumerator = "".GetEnumerator()) { }     => UsingStatementSyntax             (will carry on)
+            var elseHasUsingStatementLocalDeclaration = GetElseClauseChildren(elseClause)
+                .OfType<LocalDeclarationStatementSyntax>()
+                .Any(localDeclaration => localDeclaration.UsingKeyword.IsKind(SyntaxKind.UsingKeyword));
+            if (elseHasUsingStatementLocalDeclaration)
                 return;
 
-            if (!result.EndPointIsReachable || result.ExitPoints.Any(ep => IsDirectAccess(ifStatementSyntax, ep)))
+            // If there are conflicting local (variable or function) declarations in 'if' and 'else' blocks, return
+            var thenLocalIdentifiers = FindLocalIdentifiersIn(thenStatement);
+            var elseLocalIdentifiers = FindLocalIdentifiersIn(elseStatement);
+            if (thenLocalIdentifiers.Intersect(elseLocalIdentifiers).Any())
+                return;
+
+            var controlFlowAnalysis = context.SemanticModel.AnalyzeControlFlow(thenStatement);
+            if (!controlFlowAnalysis.Succeeded)
+                return;
+
+            if (!controlFlowAnalysis.EndPointIsReachable || controlFlowAnalysis.ExitPoints.Any(ep => IsDirectAccess(ifStatement, ep)))
             {
-                context.ReportDiagnostic(s_rule, elseClauseSyntax.ElseKeyword);
+                context.ReportDiagnostic(s_rule, elseClause.ElseKeyword);
             }
         }
 
+        internal static IEnumerable<SyntaxNode> GetElseClauseChildren(ElseClauseSyntax elseClauseSyntax)
+        {
+            return elseClauseSyntax.Statement is BlockSyntax elseBlockSyntax ?
+                elseBlockSyntax.ChildNodes() :
+                new[] { elseClauseSyntax.Statement };
+        }
+
+        private static IEnumerable<string> FindLocalIdentifiersIn(SyntaxNode node)
+        {
+            return node
+                .DescendantNodes()
+                .Where(node => node.IsKind(SyntaxKind.VariableDeclarator) || node.IsKind(SyntaxKind.LocalFunctionStatement))
```

Comment:
Does it detect the following pattern?

````
if (a is bool b) // Is 'b' a VariableDeclarator
````

Reply:
Now it does. :)